lib/tlog/session.c

```diff
@@ -99,45 +99,36 @@ tlog_session_format_lock_path(char **ppath, unsigned int id)
-            goto cleanup;                                       \
-        }                                                       \
-                                                                \
-        /* Restore EGID */                                      \
-        if (setegid(_orig_egid) < 0) {                          \
-            grc = TLOG_GRC_ERRNO;                               \
-            tlog_errs_pushc(__perrs, grc);                      \
-            tlog_errs_pushs(__perrs, "Failed restoring EGID");  \
-            goto cleanup;                                       \
-        }                                                       \
+    do {                                                                 \
+        uid_t _orig_euid = geteuid();                                    \
+        gid_t _orig_egid = getegid();                                    \
+                                                                         \
+        /* Set EUID temporarily */                                       \
+        if (seteuid(_euid) < 0) {                                        \
+            grc = TLOG_GRC_ERRNO;                                        \
+            TLOG_ERRS_RAISECS(grc, "Failed setting EUID");               \
+        }                                                                \
+                                                                         \
+        /* Set EGID temporarily */                                       \
+        if (setegid(_egid) < 0) {                                        \
+            grc = TLOG_GRC_ERRNO;                                        \
+            TLOG_ERRS_RAISECS(grc, "Failed setting EGID");               \
+        }                                                                \
+                                                                         \
+        /* Evaluate */                                                   \
+        _expr;                                                           \
+                                                                         \
+        /* Restore EUID */                                               \
+        if (seteuid(_orig_euid) < 0) {                                   \
+            grc = TLOG_GRC_ERRNO;                                        \
+            TLOG_ERRS_RAISECS(grc, "Failed restoring EUID");             \
+        }                                                                \
+                                                                         \
+        /* Restore EGID */                                               \
+        if (setegid(_orig_egid) < 0) {                                   \
+            grc = TLOG_GRC_ERRNO;                                        \
+            TLOG_ERRS_RAISECS(grc, "Failed restoring EGID");             \
+        }                                                                \
```

Comment:
I think this is wrong. It's not using the __perrs pointer. We should either leave it as is, or remove the `_perrs` argument from the macro. I think the latter might be better. What do you think?